elastic/kibana
# [Logs Explorer] Hide hard coded flyout actions using customization extension point

## 📓 Summary

Closes #165217 

This work implements a new extension point in the Discover customization framework to allow customizing the flyout content. Although it enables customizing only the actions displayed on top, it could be enhanced for additional flyout customizations in the future.

To keep it simple and familiar to other customizations, it relies on a similar API as the `top_nav` customization, allowing one to disable default actions and insert additional ones.

```ts
/**
 * Hide flyout actions to prevent rendering hard-coded actions.
 */
customizations.set({
  id: 'flyout',
  actions: {
    defaultActions: {
      viewSingleDocument: { disabled: true },
      viewSurroundingDocument: { disabled: true },
    },
  },
});
```

## 🧪 Testing 
- Navigate to `/app/observability-log-explorer` and expand the flyout for any grid entry. 
The actions on top of the flyout should not be displayed, only the title and the pagination control should appear.
- Navigate to `/app/discover` and expand the flyout for any grid entry. 
The actions for viewing a single documents or surrounding documents should be displayed.
